### PR TITLE
[Breaking change] Target Java 11

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Target Java 11

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

I also verified locally after doing a clean compilation, bot for Java and Scala.

## Complete description
Target Java 11 and dropping Java 8 support.

Note that for Scala 2.12 we are still targeting Java 8 for backward compatibility.

This page lists JDK compatibility of Scala: https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html

## Tracking Issue

Closes https://github.com/flyteorg/flyte/issues/3147

## Follow-up issue
_NA_

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -629,7 +633,7 @@
         <plugin>
           <groupId>net.alchim31.maven</groupId>
           <artifactId>scala-maven-plugin</artifactId>
-          <version>4.5.6</version>
+          <version>4.7.2</version>
```
Comment (honnix) [inline pom.xml:636]: This enables targeting Java 11 for Scala 2.13.

**Review by narape — APPROVED** [review]
